File: tourism-pojo/src/main/java/com/qinghuan/pojo/entity/Venue.java
package com.qinghuan.pojo.entity;

import com.qinghuan.pojo.enums.VenueStatus;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
public class Venue extends BaseEntity {

    private String name;
    private String address;
    private String description;
    private String coverObjectKey;
    private VenueStatus status;
    /** 景点经度，Redis GEO 写入时必须放在纬度前面。 */
    private BigDecimal longitude;
    /** 景点纬度。 */
    private BigDecimal latitude;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/operator/src/types/venue.ts
export type VenueStatus = 'ENABLED' | 'DISABLED'

export interface Venue {
  id: number
  name: string
  address: string
  description: string | null
  coverUrl: string | null
  status: VenueStatus
  longitude: number | null
  latitude: number | null
}

export interface VenueUpdateRequest {
  name: string
  address: string
  description: string
  status: VenueStatus
  longitude: number | null
  latitude: number | null
  coverImage?: File
}
